tools/docker-network/tests/dockerframework.go

```diff
@@ -368,435 +352,304 @@ func (d *DockerTestFramework) StopIssueCandidacyPayload(nodes ...*Node) {
+// PrepareBlockIssuance prepares the BlockIssuance and Congestion response, and increase BIC of the issuer if necessary.
+func (d *DockerTestFramework) PrepareBlockIssuance(ctx context.Context, clt *nodeclient.Client, issuerAddress *iotago.AccountAddress, skipBICCheck ...bool) (*api.IssuanceBlockHeaderResponse, *api.CongestionResponse) {
+	skipCheck := false
+	if len(skipBICCheck) > 0 {
+		skipCheck = skipBICCheck[0]
+	}
 
 	issuerResp, err := clt.BlockIssuance(ctx)
 	require.NoError(d.Testing, err)
 
-	congestionResp, err := clt.Congestion(ctx, from.AccountAddress, 0, lo.PanicOnErr(issuerResp.LatestCommitment.ID()))
-	require.NoError(d.Testing, err)
-
-	signedTx, err := builder.NewTransactionBuilder(apiForSlot, signer).
-		AddInput(&builder.TxInput{
-			UnlockTarget: fundsAddr,
-			InputID:      fundsOutputID,
-			Input:        fundsUTXOOutput,
-		}).
-		AddInput(&builder.TxInput{
-			UnlockTarget: from.AccountOutput.UnlockConditionSet().Address().Address,
-			InputID:      from.OutputID,
-			Input:        from.AccountOutput,
-		}).
-		AddOutput(accTransitionOutput).
-		AddOutput(foundryOutput).
-		SetCreationSlot(currentSlot).
-		AddBlockIssuanceCreditInput(&iotago.BlockIssuanceCreditInput{AccountID: from.AccountID}).
-		AddCommitmentInput(&iotago.CommitmentInput{CommitmentID: lo.Return1(issuerResp.LatestCommitment.ID())}).
-		AllotAllMana(currentSlot, from.AccountID, 0).
-		Build()
+	congestionResp, err := clt.Congestion(ctx, issuerAddress, 0, lo.PanicOnErr(issuerResp.LatestCommitment.ID()))
 	require.NoError(d.Testing, err)
 
-	blkID := d.SubmitPayload(ctx, signedTx, wallet.NewEd25519Account(from.AccountID, from.BlockIssuerKey), congestionResp, issuerResp)
-
-	updatedAccount = &Account{
-		AccountID:      from.AccountID,
-		AccountAddress: from.AccountAddress,
-		BlockIssuerKey: from.BlockIssuerKey,
-		AccountOutput:  accTransitionOutput,
-		OutputID:       iotago.OutputIDFromTransactionIDAndIndex(lo.PanicOnErr(signedTx.Transaction.ID()), 0),
+	if skipCheck {
+		return issuerResp, congestionResp
 	}
 
-	d.AwaitTransactionPayloadAccepted(ctx, blkID)
-
-	fmt.Println("Create native tokens transaction sent, blkID:", blkID.ToHex(), ", txID:", lo.PanicOnErr(signedTx.Transaction.ID()).ToHex(), ", slot:", blkID.Slot())
-
-	// wait for the account to be committed
-	d.AwaitCommitment(blkID.Slot())
-
-	d.AssertIndexerAccount(updatedAccount)
-	d.AssertIndexerFoundry(foundryID)
-
-	return updatedAccount
-}
-
-func (d *DockerTestFramework) CheckAccountStatus(ctx context.Context, blkID iotago.BlockID, txID iotago.TransactionID, creationOutputID iotago.OutputID, accountAddress *iotago.AccountAddress, checkIndexer ...bool) {
-	// request by blockID if provided, otherwise use txID
-	// we take the slot from the blockID in case the tx is created earlier than the block.
-	clt := d.Node("V1").Client
-	slot := blkID.Slot()
+	if congestionResp.BlockIssuanceCredits <= 500 {
```

Comment:
We should use RMC from the congestion response and the block to calculate this exactly.

Reply:
We should use the user the same flow as the block issuer in the mock package https://github.com/iotaledger/iota-core/blob/28a4e44108558839f4238ebbd2d1676cefdbf7bf/pkg/testsuite/mock/blockissuer.go#L214-L288

Reply:
FOr now, I would actually just remove the skipCheck option and the IncreaseBIC function as a way of increasing BIC because it is not how a user would do it. The additional required BIC should be allotted in the same transaction by providing an addiitonal input if it is required to increase BIC, not separate transaction for the sole purpose of increasing BIC.